src/lib/Form/Type/ContentType/ContentTypeEditType.php

```diff
@@ -0,0 +1,180 @@
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\Exception\AlreadySubmittedException;
+use Symfony\Component\Form\Exception\LogicException;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ContentTypeEditType extends AbstractType
+{
+    /** @var \eZ\Publish\API\Repository\LanguageService */
+    protected $languageService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    /**
+     * @param \eZ\Publish\API\Repository\LanguageService $languageService
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     */
+    public function __construct(
+        LanguageService $languageService,
+        ContentTypeService $contentTypeService
+    ) {
+        $this->languageService = $languageService;
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add(
+                'content_type',
+                ContentTypeType::class,
+                [
+                    'label' => false,
+                    'attr' => [
+                        'hidden' => true,
+                    ],
+                ]
+            )
+            ->add(
+                'content_type_group',
+                ContentTypeGroupType::class
+            )
+            ->add(
+                'add',
+                SubmitType::class,
+                [
+                    'label' => /** @Desc("Create") */ 'content_translation_add_form.add',
+                ]
+            )
+            ->addEventListener(FormEvents::PRE_SET_DATA, [$this, 'onPreSetData'])
+            ->addEventListener(FormEvents::PRE_SUBMIT, [$this, 'onPreSubmit']);
+    }
+
+    /**
+     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'data_class' => ContentTypeEditData::class,
+            'translation_domain' => 'forms',
+        ]);
+    }
+
+    /**
+     * Adds language fields and populates options list based on default form data.
+     *
+     * @param FormEvent $event
+     *
+     * @throws UnauthorizedException
+     * @throws NotFoundException
+     * @throws AlreadySubmittedException
+     * @throws LogicException
+     * @throws UnexpectedTypeException
+     */
+    public function onPreSetData(FormEvent $event)
+    {
+        $contentLanguages = [];
```

Comment:
Just a question - can't you pass `ContentType` as a form option and build the form in `buildForm` method without touching event listeners?